Create where.md entry (#6211)
Codecademy/docs: content/pytorch/concepts/tensor-operations/terms/where/where.md

---
Title: '.where()'

> Description: 'Returns a new tensor with elements selected from either `input` or `other`, depending on the condition.'

Reviewer: ```suggestion
Description: 'Returns values from one tensor if a condition is true; otherwise, it takes values from another tensor.'
```

Reviewer: Since we haven't syntax, having `input` or `other` could be confusing, so reframed it for better readability.

Subjects:
  - 'Computer Science'
  - 'Machine Learning'
Tags:

>   - 'Python'
>   - 'Machine Learning'
>   - 'Functions'
>   - 'Tensor'

Reviewer: ```suggestion
  - 'Functions'
  - 'Machine Learning'
  - 'Python'
  - 'Tensor'
```

Reviewer: arranged in alphabetical order

CatalogContent:
  - 'intro-to-py-torch-and-neural-networks'
  - 'paths/computer-science'
  - 'paths/machine-learning'
---

> In PyTorch, the **`.where()`** function returns a new tensor with elements selected from either `input` or `other`, depending on the condition. The function is part of the `torch` module.

Reviewer: ```suggestion
In PyTorch, the **`.where`** function returns a new [tensor](https://www.codecademy.com/resources/docs/pytorch/tensors) with elements chosen based on a condition, selecting values from one source when the condition is met and from another when it is not.
```

## Syntax

```pseudo

> torch.where(condition, x, y) → Tensor

Reviewer: ```suggestion
torch.where(condition, input, other) → Tensor
```

```

> - `condition`: The condition tensor.

Reviewer: ```suggestion
- `condition`: The boolean tensor that controls selection.
```

> - `x`: The tensor to select elements from when the condition is `True`.
> - `y`: The tensor to select elements from when the condition is `False`.

Reviewer: ```suggestion
- `input`: The tensor to select elements from when the condition is `True`.
- `other`: The tensor to select elements from when the condition is `False`.
```

> ## Returns
> 
> A tensor of elements selected from either `x` or `y`, depending on the condition.

Reviewer: ```suggestion
It returns a tensor of elements selected from either `input` or `other`, based on the condition.
```

## Example

The following example demonstrates the usage of the `.where()` function:

```py
import torch

# Define tensors
condition = torch.tensor([[True, False], [False, True]])
x = torch.tensor([[1, 2], [3, 4]])
y = torch.tensor([[9, 8], [7, 6]])

# Select elements based on the condition
res = torch.where(condition, x, y)

# Print the result
print(res)
```

The above code produces the following output:

```shell
tensor([[1, 8],
        [7, 4]])
```

In this example, the `.where()` function selects elements from the `x` tensor where the condition tensor is `True` and from the `y` tensor where the condition tensor is `False`. The resulting tensor is created by selecting elements from `x` or `y` based on the condition tensor.

> The `.where()` function is particularly useful in conditional operations where you need to select elements from different tensors based on a condition.

Reviewer: ```suggestion
The `.where()` function is particularly useful in conditional operations where element-wise selection between tensors is required.
```
